integration/common/openlineage/common/provider/dbt/local.py

```diff
@@ -101,10 +104,45 @@ def __init__(
 
         self.target = target
         self.project_name = dbt_project["name"]
+        self.models = models or []
+        self.selector = selector
         self.profile_name = profile_name or dbt_project.get("profile")
         if not self.profile_name:
             raise KeyError(f"profile not found in {dbt_project}")
 
+    @property
+    def _use_extended_job_name(self):
+        # The name of the job published on the OpenLineage events for the dbt-ol run wrapper
+        # should include extended attributes (profile and model) only if this environment
+        # variable is set, in order to prevent breaking back-compatibility.
+        var = os.getenv("OPENLINEAGE_DBT_USE_EXTENDED_JOB_NAME", "false").lower()
```

Comment:
@mobuchowski @JDarDagran do you think it makes sense to control the job name structure using this environment variable (I introduced it to avoid the case where a job is named `A` before the update and `B` afterwards), or is it ok to just apply the new logic to all the workflows?

Reply:
@blacklight any idea whether we might want to use enum here rather than boolean? Like, explicitely choosing between job names depending on `model` or `selector` rather than forcing the hierarchy in `job_name` function?

Reply:
That would make sense if dbt supports a mix of models list + model selector, but I haven't toyed enough with dbt to know if that is the case.

My understanding is that dbt supports either a list of models, or a model selector, but not both (at least not in the cases I've seen).

If that is the case then I don't see much value in an enum approach - either the user wants an extended job name or not, and if so then we already know which components to expect.

@JDarDagran do you know if a model+selector approach is supported?

Reply:
`dbt run --target dev --select tag:nightly --selector nightly` this simple example seems to be skipping `--select tag:nightly` if passing `--selector`

Reply:
>If that is the case then I don't see much value in an enum approach - either the user wants an extended job name or not, and if so then we already know which components to expect.

I'm okay with that